system/security/shared/caching.cpp

```diff
@@ -537,6 +518,33 @@ inline void CPermissionsCache::removeAllManagedFileScopes()
     m_managedFileScopesMap.clear();
 }
 
+static CriticalSection setCacheTimeoutCriticalSection;
+void  CPermissionsCache::setCacheTimeout(int timeoutSeconds)
+{
+    CriticalBlock setCacheTimeoutBlock(setCacheTimeoutCriticalSection);
+    m_cacheTimeoutInSeconds = timeoutSeconds;
+
+    if(m_cacheTimeoutInSeconds == 0 && isTransactionalEnabled())//ensure transactional time is updated
+        setTransactionalCacheTimeout(DEFAULT_CACHE_TIMEOUT_SECONDS); //Transactional timeout is set to 10 seconds for long transactions that might take over 10 seconds.
+    else
+        setTransactionalCacheTimeout(timeoutSeconds);
+
+    if (m_secMgr)
```

Comment:
it would be better to pass the ISecManager through at ctor time (and through CPermissionsCache::getInstance), and remove need for setSecManager, and assert that a CPermissionsCache doesn't exist with a null m_secMgr, so remove this condition too.